src/plugins/kibana_utils/public/state_containers/create_state_container.ts

```diff
@@ -56,9 +74,13 @@ export const createStateContainer = <
     state$,
     getState: () => data$.getValue(),
     set: (state: State) => {
-      data$.next(freeze(state));
+      container.dispatch({ type: $$setActionType, args: [state] });
```

Comment:
this makes `set()` to go through dispatch and not be skipped by added middlewares